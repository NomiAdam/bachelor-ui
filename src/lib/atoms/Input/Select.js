// eslint-disable-next-line no-unused-vars
import React from 'react';
import styled from 'styled-components';
import { basicTheme } from '../../constants/theme';

export const StyledSelect = styled.select`
	height: 30px;
  	min-width: 50px;
  	padding: .25em;
  	border: 0 solid transparent;
  	border-radius: 3px;
  	border-bottom: ${ ({ withBorder }) => (withBorder ? `2px solid ${ basicTheme.blue }` : '0px') };
  	overflow-y: scroll;
  	font-weight: bold;
  	letter-spacing: .15em;
  	${ ({ error }) => (error ? `
	    border: 1px solid ${ basicTheme.red };
	    border-top-left-radius: 5px;
        border-top-right-radius: 5px;
	` : '') }
  	&:focus {
  		outline: none;
  	}
  	box-shadow: 1px 2px 10px -2px ${ basicTheme.grey };
`;

export default StyledSelect;
