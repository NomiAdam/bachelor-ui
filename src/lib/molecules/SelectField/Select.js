import React from 'react';
import PropTypes from 'prop-types';
import Option from './Option';
import styled from 'styled-components';
import {map,length,equals,o} from 'ramda';
import { darkTheme, lightTheme } from '../../constants/theme';

const StyledSelect = styled.div`
    overflow: hidden;
    display: ${props => props.visible ? 'flex' : 'none' }
    top: -50px;
    position: absolute;
    z-index: 999;
    background-color: ${lightTheme.white};
    width: ${({width}) => width};
    min-height: 0px;
    max-height: 500px;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    border: 1px solid transparent; 
    border-radius: 3px;
    ${props => props.isEmpty ? 'border: none;' : ''}
    font-size: 16px;
  	box-shadow: 5px 5px 15px ${darkTheme.darkGrey};
`;

const doesItHaveOptions = o(equals(0),length);
const Select = ({visible, options, onClick, width, selected}) => {
    const equalsSelected = equals(selected);
    const mapSuggestion = ({value, label}) => (
        <Option value={value} label={label} onClick={onClick} key={value} selected={equalsSelected(value)} />
    );
    const mappedOptions = map(mapSuggestion);
    return (
        <StyledSelect visible={visible} isEmpty={doesItHaveOptions(options)} width={width}>
            {visible && mappedOptions(options)}
        </StyledSelect>
    );
};

Select.propTypes = {
    /**
     * Whether select div is visible or not, is visible when Input is focused
     */
    visible: PropTypes.bool,
    /**
     * Array of possible values [ { value: 'Value', label: 'Label' }, ... ]
     */
    options: PropTypes.array,
    /**
     * Option onMouseDown handler
     */
    onClick: PropTypes.func,
    /**
     * String representation of select width
     */
    width: PropTypes.string,
    /**
     * String definition of currently selected item
     */
    selected: PropTypes.string,
};

Select.defaultProps = {
    options: [],
};

export { StyledSelect };
export default Select;
