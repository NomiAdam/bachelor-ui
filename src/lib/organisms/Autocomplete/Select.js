import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
    map, length, equals, o,
} from 'ramda';
import Option from './Option';
import { lightTheme } from '../../constants/theme';

const StyledSelect = styled.div`
    display: ${ props => (props.visible ? 'block' : 'none') }
    position: absolute;
    z-index: 999;
    background-color: ${ lightTheme.white };
    width: 100%;
    min-height: 0px;
    max-height: 500px;
    border: 1px solid ${ lightTheme.lightGrey }; 
    border-top: none;
    border-bottom-right-radius: 4px;
    border-bottom-left-radius: 4px;
    ${ props => (props.isEmpty ? 'border: none;' : '') }
    font-weight: 300;
    font-size: 16px;
    overflow-y: auto;
`;

const doesItHaveOptions = o(equals(0), length);
const Select = ({
    visible, suggestions, onClick, selected,
}) => {
    const equalsSelected = equals(selected);
    const mapSuggestion = ({ value }) => (
        <Option value={ value } onClick={ onClick } key={ value } selected={ equalsSelected(value) } />
    );
    const mappedOptions = map(mapSuggestion);
    return (
        <StyledSelect visible={ visible } isEmpty={ doesItHaveOptions(suggestions) }>
            {visible && mappedOptions(suggestions)}
        </StyledSelect>
    );
};

Select.propTypes = {
    /**
     * Whether select div is visible or not, is visible when Input is focused
     */
    visible: PropTypes.bool,
    /**
     * Array of possible values [ { value: 'Value', label: 'Label' }, ... ]
     */
    suggestions: PropTypes.array,
    /**
     * Option onMouseDown handler
     */
    onClick: PropTypes.func,
    /**
     * String definition of selected suggestion value
     */
    selected: PropTypes.string,
};

Select.defaultProps = {
    suggestions: [],
};

export { StyledSelect };
export default Select;
