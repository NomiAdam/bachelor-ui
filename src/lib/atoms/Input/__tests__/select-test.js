/* eslint-disable no-undef */
import React from 'react';
import {shallow} from 'enzyme';
import Select from '../Select';

describe('Select', () => {
	it('should be defined', () => {
		expect(shallow(<Select/>)).toBeDefined();
	});
	it('should match the snapshot', () => {
		expect(shallow(<Select/>)).toMatchSnapshot();
	});
});