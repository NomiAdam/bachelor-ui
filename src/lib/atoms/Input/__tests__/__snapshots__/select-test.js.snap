// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should match the snapshot 1`] = `
<select
  className="sc-bdVaJa jxhoIt"
/>
`;
